dynamiqs/integrators/core/rouchon_integrator.py

```diff
@@ -48,82 +51,160 @@ def step(self, terms, t0, t1, y0, args, solver_state, made_jump):
-            I = eye_like(H)
-            LdL = sum([_L.dag() @ _L for _L in L])
-
-            M0 = I - (1j * H + 0.5 * LdL) * delta_t
-            Ms = [jnp.sqrt(delta_t) * _L for _L in L]
+            t = (t0 + t1) / 2
+            dt = t1 - t0
+            Ms = self._kraus_ops(t, dt)
 
             if self.method.normalize:
-                rho = cholesky_normalize(M0, LdL, delta_t, rho)
+                rho = cholesky_normalize(Ms, rho)
 
-            return M0 @ rho @ dag(M0) + sum([_M @ rho @ dag(_M) for _M in Ms])
+            return sum([M @ rho @ M.dag() for M in Ms])
 
         return AbstractRouchonTerm(kraus_map)
 
+    @abstractmethod
+    def _kraus_ops(self, t: float, dt: float) -> list[QArray]:
+        pass
+
+
+class MESolveRouchon1Integrator(MESolveRouchonIntegrator):
+    """Integrator computing the time evolution of the Lindblad master equation using the
+    Rouchon 1 method.
+    """
+
+    def _kraus_ops(self, t: float, dt: float) -> list[QArray]:
+        # M0 = I - (iH + 0.5 sum_k Lk^† @ Lk) dt
+        # Mk = Lk sqrt(dt)
+        L, H = self.L(t), self.H(t)
+        LdL = sum([_L.dag() @ _L for _L in L])
+        G = -1j * H - 0.5 * LdL
+        e1 = (dt * G).expm() if self.method.exact_expm else _expm_taylor(dt * G, 1)
+        return [e1] + [jnp.sqrt(dt) * _L for _L in L]
+
 
 mesolve_rouchon1_integrator_constructor = lambda **kwargs: MESolveRouchon1Integrator(
-    **kwargs, diffrax_solver=Rouchon1DXSolver(), fixed_step=True
+    **kwargs, diffrax_solver=RouchonDXSolver(1), fixed_step=True
+)
+
+
+class MESolveRouchon2Integrator(MESolveRouchonIntegrator):
+    """Integrator computing the time evolution of the Lindblad master equation using the
+    Rouchon 2 method.
+    """
+
+    def _kraus_ops(self, t: float, dt: float) -> list[QArray]:
+        L, H = self.L(t), self.H(t)
+        LdL = sum([_L.dag() @ _L for _L in L])
+        G = -1j * H - 0.5 * LdL
+        e1 = (dt * G).expm() if self.method.exact_expm else _expm_taylor(dt * G, 2)
+
+        return (
+            [e1]
+            + [jnp.sqrt(dt / 2) * e1 @ _L for _L in L]
+            + [jnp.sqrt(dt / 2) * _L @ e1 for _L in L]
+            + [jnp.sqrt(dt**2 / 2) * _L1 @ _L2 for _L1, _L2 in product(L, L)]
+        )
+
+
+mesolve_rouchon2_integrator_constructor = lambda **kwargs: MESolveRouchon2Integrator(
+    **kwargs, diffrax_solver=RouchonDXSolver(2), fixed_step=True
+)
+
+
+class MESolveRouchon3Integrator(MESolveRouchonIntegrator):
+    """Integrator computing the time evolution of the Lindblad master equation using the
+    Rouchon 3 method.
+    """
+
+    def _kraus_ops(self, t: float, dt: float) -> list[QArray]:
+        L, H = self.L(t), self.H(t)
+        LdL = sum([_L.dag() @ _L for _L in L])
+        G = -1j * H - 0.5 * LdL
+        e1o3 = (
+            (dt / 3 * G).expm()
+            if self.method.exact_expm
+            else _expm_taylor(dt / 3 * G, 3)
+        )
+        e2o3 = e1o3 @ e1o3
+        e3o3 = e2o3 @ e1o3
```

Comment:
I'm mildly surprised that there's no equivalent of this in the second-order Rouchon method, i.e. of requiring `e1o2` and `e2o2` in the equations. Are you sure about the equations? Did you check that these methods are indeed 2nd and 3rd order by numerically varying dt and observing a dt^2 or dt^3 scaling?